atlasdb-conjure/src/main/java/com/palantir/atlasdb/http/v2/ConjureJavaRuntimeTargetFactory.java

```diff
@@ -76,19 +76,16 @@ private ConjureJavaRuntimeTargetFactory() {
 
     @Override
     public <T> InstanceAndVersion<T> createLiveReloadingProxyWithFailover(
-            Supplier<ServerListConfig> serverListConfigSupplier,
+            Refreshable<ServerListConfig> serverListConfigRefreshable,
             Class<T> type,
             AuxiliaryRemotingParameters parameters) {
         ClientOptions options = getClientOptionsForFailoverProxy(parameters);
-        Supplier<T> clientSupplier = new CachedTransformingSupplier<>(
-                serverListConfigSupplier,
-                serverListConfig -> JaxRsClient.create(
-                        type,
-                        addAtlasDbRemotingAgent(parameters.userAgent()),
-                        NoOpHostEventsSink.INSTANCE,
-                        options.serverListToClient(serverListConfig)));
-
-        return decorateFailoverProxy(type, clientSupplier);
+        Refreshable<T> refreshableClient = serverListConfigRefreshable.map(serverListConfig -> JaxRsClient.create(
```

Comment:
Why do we want to keep `CachedTransformingSupplier` around? Seems like we should always take the Refreshable path?

Reply:
🎉